flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -165,7 +165,8 @@ public void submitApplicationCluster(
         if (requireHaMetadata) {
             validateHaMetadataExists(conf);
         }
-        deployApplicationCluster(jobSpec, conf);
+
+        deployApplicationCluster(jobSpec, removeOperatorConfigs(conf));
```

Comment:
I think you might have forgotten adding it for session cluster deployments.

Reply:
Do you mean submitSessionCluster in NativeFlinkService? kubernetesClusterDescriptor.deploySessionCluster takes cluster specification as input. It already helped to filter unnecessary configs?

Reply:
Yes I think this needs to be added to the `submitSessionCluster` implementations for both the native and standalone service.

A proper test would actually validate all deployment modes and make sure the config is filtered :
  Native:
   - Application
   - Session
  Standalone
   - Application
   - Session
  
  + SessionJob

Reply:
Thanks for the suggestions. Test cases are added in the latest revision.